src/routes/app/wallets/_components/AmountOfCoins.tsx

```diff
@@ -17,84 +19,101 @@ export default component$<AmountOfCoinsProps>(
     });
     return (
       <>
-        <div class="mb-8">
-          <Label name="Select tokens" />
-          {addWalletFormStore.coinsToCount.map((symbol) => (
-            <div class="flex max-h-[500px] flex-col overflow-auto" key={symbol}>
-              <div class="flex items-center justify-between gap-2">
-                <FormBadge
-                  key={symbol}
-                  image={`/assets/icons/tokens/${symbol.toLowerCase()}.svg`}
-                  description={symbol}
-                  class="w-[98%] flex-row-reverse"
-                  hasImg={"/assets/icons/dashboard/success.svg?jsx"}
-                >
-                  {" "}
-                  <input
-                    type="text"
-                    name={`${symbol}Amount`}
-                    class={`absolute end-2 z-10 block h-8 min-w-max rounded border border-[#24a148] bg-transparent p-3 pr-6 text-sm text-[#24a148] placeholder-white placeholder-opacity-50 `}
-                    placeholder={`${symbol} approval limit`}
-                    value={
-                      addWalletFormStore.coinsToApprove.find(
-                        (item) => item.symbol === symbol,
-                      )!.amount
-                    }
-                    onInput$={(e) => {
-                      const target = e.target as HTMLInputElement;
-                      // const regex = /^[0-9]*(\.[0-9]*)?$/;
-                      const regex = /^\d*\.?\d*$/;
-                      target.value = replaceNonMatching(
-                        target.value,
-                        regex,
-                        "",
+        <div class="flex flex-col gap-2 pb-3">
+          <Label name="Selected tokens" />
+          {addWalletFormStore.coinsToCount.map((symbol) => {
+            const coin = addWalletFormStore.coinsToApprove.find(
+              (item) => item.symbol === symbol,
+            );
+
+            const isValid = () => {
+              return (
+                checkPattern(coin!.amount, /^\d*\.?\d*$/) &&
+                coin!.amount != "0" &&
+                coin!.amount.length > 0 &&
+                coin!.amount[0] != "0"
+              );
+            };
```

Comment:
I'd rather use simple if condition to check if coin and coint.amount is not null e.g.
```
if(coin && coin.amount) {
rerturn (...
```
In this case we dont have to use 
```
coin!.amount
``` 
anywhere.
IMO thast's more readable but you can ask other developers for opinion e.g. @pgajek

Reply:
Fixed